tests/command.bats

```diff
@@ -33,6 +33,34 @@ load "$BATS_PATH/load.bash"
   unstub docker
 }
 
+@test "can pass through optional params" {
+  export BUILDKITE_PLUGIN_JUNIT_ANNOTATE_ARTIFACTS="junits/*.xml"
+  export BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN="custom_(*)_pattern.xml"
+  export BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT="file"
+
+  artifacts_tmp="tests/tmp/$PWD/junit-artifacts"
+  annotation_tmp="tests/tmp/$PWD/junit-annotation"
+
+  stub mktemp \
+    "-d junit-annotate-plugin-artifacts-tmp.XXXXXXXXXX : mkdir -p $artifacts_tmp; echo $artifacts_tmp" \
+    "-d junit-annotate-plugin-annotation-tmp.XXXXXXXXXX : mkdir -p $annotation_tmp; echo $annotation_tmp"
+
+  stub buildkite-agent "artifact download junits/*.xml /plugin/tests/tmp//plugin/junit-artifacts : echo Downloaded artifacts" \
+                       "annotate --context junit --style error : echo Annotation added"
+
+  stub docker "--log-level error run --rm --volume /plugin/tests/tmp//plugin/junit-artifacts:/junits --volume /plugin/hooks/../ruby:/src --env "BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN=${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN:-}" --env "BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT=${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT:-}" ruby:2.5-alpine /src/bin/annotate /junits : echo '<details>Failure</details>'"
```

Comment:
Is it better to stub the docker commands using `--env "BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT=${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT:-}"` vs `--env BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT=` to mimic what's happening in the wild? If so, I can update the other tests.

Reply:
Nice additional test 👌🏼

Hrmm… I think you're right with the last suggestion, in that the test shouldn't be do all the env interpolation, and should just have the the exact strings that are expected. So instead of:
`--env "BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN=${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN:-}"`
the test should probably just be:
`--env "BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN=custom_(*)_pattern.xml"`

It'd be great if you could update the others to match that too! 🙏🏼

Let me know if you need anything.

Reply:
Oh, I think you might need to check the double quotes in there too!

Inside the string starting with `"--log-level ...` you don't want any double quotes inside the string. It's a bit of a weird quirk of BATS stubbing… I do that all the time also, then remember to remove the extra double quotes. I think it becomes something like:

```bash
stub docker "--log-level error run --rm --volume /plugin/tests/tmp//plugin/junit-artifacts:/junits --volume /plugin/hooks/../ruby:/src --env BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN=custom_(*)_pattern.xml --env BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_OUTPUT=junits/*.xml ruby:2.5-alpine /src/bin/annotate /junits : echo '<details>Failure</details>'"
```